services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/document/DefaultEnhancedDocumentTest.java

```diff
@@ -0,0 +1,332 @@
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.core.SdkNumber;
+import software.amazon.awssdk.enhanced.dynamodb.AttributeConverterProvider;
+import software.amazon.awssdk.enhanced.dynamodb.DefaultAttributeConverterProvider;
+import software.amazon.awssdk.enhanced.dynamodb.EnhancedType;
+import software.amazon.awssdk.enhanced.dynamodb.internal.converter.ChainConverterProvider;
+import software.amazon.awssdk.enhanced.dynamodb.internal.document.DefaultEnhancedDocument;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+import software.amazon.awssdk.utils.Pair;
+
+public class DefaultEnhancedDocumentTest {
+
+    public static final String SIMPLE_NUMBER_KEY = "numberKey";
+    public static final String BIG_DECIMAL_NUMBER_KEY = "bigDecimalNumberKey";
+    public static final String BOOL_KEY = "boolKey";
+    public static final String NULL_KEY = "nullKey";
+    public static final String NUMBER_SET_KEY = "numberSet";
+    public static final String SDK_BYTES_SET_KEY = "sdkBytesSet";
+    public static final String STRING_SET_KEY = "stringSet";
+    public static final String[] STRINGS_ARRAY = {"a", "b", "c"};
+    public static final SdkBytes[] SDK_BYTES_ARRAY = {SdkBytes.fromUtf8String("a"), SdkBytes.fromUtf8String("b"),
+                                                      SdkBytes.fromUtf8String("c")};
+    public static final String[] NUMBER_STRING_ARRAY = {"1", "2", "3"};
+    static final String SIMPLE_STRING = "stringValue";
+    static final String SIMPLE_STRING_KEY = "stringKey";
+    static final String SIMPLE_INT_NUMBER = "10";
+
+    private static Stream<Arguments> attributeValueMapsCorrespondingDocuments() {
+
+
+        return Stream.of(
+
+            //1. Null valuw
+            Arguments.of(
+                // {"nullKey": null}
+                map()
+                    .withKeyValue("nullKey", AttributeValue.fromNul(true)),
+                documentBuilder()
+                    .addNull("nullKey")
+                    .build(), "{" + "\"nullKey\": null" + "}"
+            ),
+
+            //2. Simple String
+            Arguments.of(
+                map()
+                    .withKeyValue(SIMPLE_STRING_KEY, AttributeValue.fromS(SIMPLE_STRING)),
+                documentBuilder()
+                    .add(SIMPLE_STRING_KEY, SIMPLE_STRING)
+                    .build(), "{" + "\"stringKey\": \"stringValue\"" + "}"
+            ),
+
+            // 3. Different Number Types
+            Arguments.of(map()
```

Comment:
Is it possible to be consistent with the placement of `map()`? It's easier to read the structure if it's the same